protodata/src/main/proto/spine/validation/rule.proto

```diff
@@ -0,0 +1,138 @@
+syntax = "proto3";
+
+package spine.validation;
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.validation";
+option java_outer_classname = "RuleProto";
+option java_multiple_files = true;
+
+import "spine/validation/value.proto";
+import "spine/protodata/ast.proto";
+import "google/protobuf/any.proto";
+
+// A simple validation rule.
+//
+// May be expressed as a comparison between two operands. The first operand is always a field value.
+// The second operand is the `other_value`. `other_value` is typically derived from the Protobuf
+// definition of the validation rule.
+//
+message Rule {
+
+    // The field which is being compared.
+    protodata.Field field = 1;
+
+    oneof operation {
+
+        // A simple comparison sign.
+        Sign sign = 2;
+
+        // A custom comparison method, which cannot be covered by a combination of `Sign`s.
+        UnknownOperation unknown_operation = 3;
+    }
+
+    // The value to which the field is compared.
+    Value other_value = 4;
+
+    // A human-readable message explaining the error.
+    //
+    // May include tokens `{value}` and `{other}`, which will be replaced with the runtime values
+    // when the error is constructed.
+    //
+    string error_message = 5;
+}
+
+// A predefined comparison sign.
+//
+// Comparison between two values of the same type must always yield a boolean result.
+//
+enum Sign {
+
+    OPERATION_UNKNOWN = 0;
+
+    EQUAL = 1;
+
+    NOT_EQUAL = 2;
+
+    LESS_THAN = 3;
+
+    GREATER_THAN = 4;
+
+    LESS_OR_EQUAL = 5;
+
+    GREATER_OR_EQUAL = 6;
+
+    DISTINCT = 7;
+}
+
+// A non-predefined comparison option.
+message UnknownOperation {
+
+    // Describes how the renderer should compare the values.
+    string description = 1;
+}
+
+message RuleOrComposite {
```

Comment:
I would name this `Rule`. Then its children would be `PlainRule` and `CompositeRule`.